examples/official-site/migrations.sql

```diff
@@ -0,0 +1,153 @@
+
+*If you are in early stages of development and are okay with losing data*, you can delete the database and start over with an altered migration file. However, in a production environment, especially once persisted data is involved, this is not an option.
+
+It's like trying to go back in time and change a previous sedimentary layer. That's not how rocks work, and that's not how migrations work.
+
+Append or amend; do not try to change the past.
+
+## Examples
+
+Let's start off easy with a simple database to store user information: `first_name`, `last_name`, `email`, `phone`, and `password_hash`. Our first migration actually creates the `users` table with these columns. That is, we migrate from *nothing* to *having a users table*.
+
+**`sqlpage/migrations/001_create_users.sql`**:
+```sql
+create table users (
+	id integer primary key autoincrement,
+	first_name not null,
+	last_name not null,
+	email not null unique,
+	phone,
+	rewards_level,
+	password_hash not null
+);
+```
+
+In the terminal, we can see the new schema:
+
+```console
+sqlite> .schema
+CREATE TABLE users (
+	id integer primary key autoincrement,
+	first_name not null,
+	last_name not null,
+	email not null unique,
+	phone,
+	rewards_level,
+	password_hash not null
+);
+```
+
+### A Simple Change
+
+Later, we discover we need a `middle_name` column, so we create a new migration file to add this column to the `users` table. Remember, we must ensure the order is written into the filename so SQLPage can apply them in the correct order when building the database.
+
+**`sqlpage/migrations/002_add_middle_name.sql`**:
+```sql
+alter table users add column middle_name;
+```
+
+In the terminal:
+
+```console
+sqlite> .schema
+CREATE TABLE users (
+	id integer primary key autoincrement,
+	first_name not null,
+	last_name not null,
+	email not null unique,
+	phone,
+	rewards_level integer,
+	password_hash not null,
+	middle_name
+);
+```
+
+### A More Complex Change
+
+But notice here, SQLite has appended the column to the very end. What if we really need that `middle_name` column to be next to the other name columns? Further, what if we realize `rewards_level` should really be an integer and only one between 1 and 20?
+
+We can create a new migration file to make these changes, albeit a bit more complicated.
+
+Because we'll be altering column types and modifying column order, we'll need to *create a temporary table* to hold the data while we drop the original table and recreate it with the new schema.
+
+**`sqlpage/migrations/003_alter_users.sql`**:
+```sql
+create table users_temp (
+	id integer primary key autoincrement,
+	first_name not null,
+	last_name not null,
+	middle_name,
+	email not null unique,
+	phone,
+	rewards_level integer check(rewards_level between 1 and 20),
+	password_hash not null
+);
+
+insert into users_temp select * from users;
```

Comment:
Maybe it would be a better practice to indicate the column names here?

Reply:
I agree, just updated 👍